mfx/vonhamos.py

```diff
@@ -0,0 +1,114 @@
+from mfx.db import mfx_von_hamos_6crystal as vh
+
+class SimpleDeterministicVH:
+    """
+    Simple wrapper for von Hamos spectrometer that provides deterministic movement capabilities.
+    
+    Examples:
+    --------
+    >>> vh = SimpleDeterministicVH()
+    
+    # Basic deterministic movement
+    >>> vh.go('c1', 'rot', 45.0)  # Move crystal 1's rotation to 45 degrees
+    
+    # Smart movement with overshooting
+    >>> vh.go('c1', 'rot', 45.0, smart=True)  # Will detect and handle stuck movement
+    
+    # Custom parameters
+    >>> vh.go('c2', 'x', 10.0, epsilon=0.001, n_iterations_max=20)  # Tighter tolerance, more attempts
+    >>> vh.go('c2', 'x', 10.0, smart=True, stuck_threshold=0.0005, overshoot_factor=1.5)  # Custom smart parameters
+    """
+
+    def __init__(self):
+        self.vh = vh
+
+    def get_motor(self, c, axis):
+        """
+        Get a specific motor.
+        
+        Parameters
+        ----------
+        c : str
+            Crystal number (e.g. 'c1', 'c2', etc.)
+        axis : str
+            Axis name (e.g. 'x', 'rot', 'tilt')
+            
+        Returns
+        -------
+        motor : BeckhoffAxis
+            The requested motor
+        """
+        return getattr(getattr(self.vh, c), axis)
+
+    def go(self, c, axis, target, epsilon=0.001, n_iterations_max=10, 
+           smart=False, stuck_threshold=0.001, overshoot_factor=1.2, wait=True):
+        """
+        Move a motor to target position with guaranteed accuracy.
+        
+        Parameters
+        ----------
+        c : str
+            Crystal number (e.g. 'c1', 'c2', etc.)
+        axis : str
+            Axis name (e.g. 'x', 'rot', 'tilt')
+        target : float
+            Target position to move to
+        epsilon : float, optional
+            Maximum allowed deviation from target, by default 0.001
+        n_iterations_max : int, optional
+            Maximum number of iterations to attempt, by default 10
+        smart : bool, optional
+            Whether to use smart movement with overshooting, by default False
+        stuck_threshold : float, optional
+            Minimum position change to consider movement not stuck, by default 0.001
+        overshoot_factor : float, optional
+            Factor to multiply error by for overshoot (should be >1 to overshoot), by default 1.2
+        wait : bool, optional
+            Whether to wait for motion to complete, by default True
+        """
+        motor = self.get_motor(c, axis)
+        low_limit, high_limit = motor.limits
+
+        if target < low_limit or target > high_limit:
+            raise ValueError(f"Target {target} is out of limits {low_limit} to {high_limit}")
```

Comment:
Nitpick: the ophyd motor objects should have a built-in for this (please excuse any typos I make here):
```suggestion
        motor.check_position(target)
```

Reply:
Nice! @MartinaDelGaudio and @amine-lamouchi , can you resolve this when you get a chance? Thank you!